test-data/unit/check-classes.test

```diff
@@ -3290,6 +3290,24 @@ x3: M = cv
 
 [builtins fixtures/classmethod.pyi]
 
+[case testMetaclassOverloadResolution]
+from typing import Type, overload
+
+class EM(type): pass
+class E(metaclass=EM): pass
+
+@overload
```

Comment:
It would be nice to have a few more test cases:

* Test two overload variants which have two different metaclasses as argument types, and overloading based on metaclass.
* Test calling the overloaded function with a `str` argument (results should be `str`).